lib/mruby/gem.rb

```diff
@@ -214,7 +214,7 @@ def generate_gem_init(fname)
           f.puts %Q[  mrb_#{funcname}_gem_init(mrb);] if objs != [objfile("#{build_dir}/gem_init")]
           unless rbfiles.empty?
             if cdump?
-              f.puts %Q[  mrb_load_proc(mrb, gem_mrblib_#{funcname}_proc);]
```

Comment:
`mrb_load_proc()` calls `mrb_vm_run)` internally, but `mrb_load_irep` calls `mrb_top_run()`. This inconsitency caused the problem. So I will fix `mrb_load_proc()` instead of this change.

Reply:
Ah, it's starting to make more sense. I appreciate the explanation, and the fix to `mrb_load_proc()`. Thanks!